src/gerbil/gxi

```diff
@@ -1,11 +1,23 @@
 #!/bin/sh
 set -eu
 
+
 if [ -z "${GERBIL_HOME:-}" ]; then
-    GERBIL_HOME=$(dirname $(cd ${0%/*} && echo $PWD))
+    SCRIPT=$(readlink --canonicalize "$0")
```

Comment:
> Beware: `--canonicalize` seems to be a Linuxism not available on Darwin or BSD. This is not acceptable here.

Yeah, knew that from the start, and was, of course, wondering who would mention it :) There's a reason why this is in a debian branch, and things like this are a major one. 

Can I just loop over readlink until it matches? I can test -L and do it for symlinks only. What do you think?

Reply:
I don't think `readlink` can be made to work portably to print a canonical path across the many variants of BSD. Maybe you'll have more luck with `realpath`, but there too you should check for its availability on Darwin, FreeBSD, NetBSD, OpenBSD as well as Linux.

We recently accepted patches to shell scripts specifically to make Gerbil run better on NetBSD, so it would not be nice to break that.

Reply:
This seems POSIX-y enough: https://github.com/drewc/gerbil/commit/1f5a2fc3e03cca8239f2916c0dc160963358e883

What do you think?